src/CMakeLists.txt

```diff
@@ -134,11 +135,18 @@ SET(LIBGD_DEP_LIBS
 	${XPM_LIBRARIES}
 	${FONTCONFIG_LIBRARY}
 	${WEBP_LIBRARIES}
+	${AVIF_LIBRARIES}
 	${RAQM_LIBRARIES}
 )
+
+if (TARGET avif)
```

Comment:
you shouldn't need this `if` block at all.  just adding `${AVIF_LIBRARIES}` above should be sufficient.  if it isn't, it means `${AVIF_LIBRARIES}` isn't set correctly.

Reply:
There is no `AVIF_LIBRARIES` variable, `avif` is a target!

Reply:
sorry, but i don't know what that means.  we should not be hardcoding `avif` as a library name.  the entire point of using pkg-config (or a cmake module) is for it to detect the right linker settings and set up a variable (like `AVIF_LIBRARIES`) for the rest of the files to use.

Reply:
Please take a look at `/usr/lib64/cmake/libavif/libavif-config-*.cmake`

The avif TARGET gets imported in the config see
`set_target_properties(avif PROPERTIES ...)`

Details on target properties are documented here:
https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#target-properties

Reply:
as long as the libavif cmake module declares `avif` such that it is expanded into the correct library for `target_link_libraries` (i.e. "A library target name"), this is fine.  but we cannot have `target_link_libraries` treat `avif` as `-lavif` using `target_link_libraries`'s own resolution logic (i.e. "A plain library name").

Reply:
i took a look at this locally.  it looks like `avif` is treated as a "a library target name" and so expands into the cmake settings.  however, it shows how fundamentally broken and/or poorly designed cmake is: it loads the full path to the libavif library and links against that directly.  this breaks fundamental cross-compiling principles.

that said, if anyone wants to use cmake, they're opting into the broken cmake design.  the autotools code cross-compiles just fine.

@morsssss you can delete these lines and change the toplevel CMakeLists.txt instead:

```
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,6 +123,8 @@ else (USE_EXT_GD)
    IF (ENABLE_AVIF)
        FIND_PACKAGE(libavif 0.8.2 REQUIRED CONFIG)
        SET(HAVE_LIBAVIF 1)
+       SET(AVIF_LIBRARIES avif)
+       SET(AVIF_FOUND 1)
    ENDIF (ENABLE_AVIF)
 
    IF (ENABLE_LIQ)
```

this builds off of the target logic that @cryptomilk wants but keeps the existing gd cmake style so we can merge this without blocking on rewriting the cmake files to use the target logic directly.

Reply:
Got it. This worked for me as well - moving all the logic to the top-level file, and letting the subsidiary `CMakeLists.txt` files use our usual flags for logic instead of targets.